### ISSUE TITLE
commandLineParser's convertJsonOption() misses check for option's null and undefined value

### ISSUE THREAD

**Issue by vvs** [microsoft/TypeScript#25148]
Take a look:
https://github.com/Microsoft/TypeScript/blob/9706f1729e9c41c9af901c279367dec0ab1093d1/src/compiler/commandLineParser.ts#L2128

The `convertJsonOption()` misses the check for `null` or `undefined` for string options, when it invokes `normalizeNonListOptionValue()`:
https://github.com/Microsoft/TypeScript/blob/9706f1729e9c41c9af901c279367dec0ab1093d1/src/compiler/commandLineParser.ts#L2137

This leads to problems in some cases. For example, when user overrides the parent's config value in his/her own tsconfig.json `declarationDir` with `null`.

`tsc` works in such case with no problems. But `gulp-typescript` complains that `declarationDir` is defined. It seems like `tsc` and `gulp-typescript` go through slightly different code paths (but both use commandLineParser API) , and `tsc` ends up with calling `normalizeOptionValue()` which has the proper check, while `gulp-typescript` goes through `convertJsonOption()` and ends up with string value, resolved to the current dir.

**Related Issues:** 
@mhegazy , this is related to what we discussed in #21443

**Comment by mhegazy** [comment]
PRs welcomed.

**Comment by a-tarasyuk** [comment]
@vvs @mhegazy 

Could you provide _expected behavior_ for this issue? I want to fix it and make PR. 

Based on the related(#21443) discussion

> ... declarationDir should be required only when declaration is true. When declaration is false, it shouldn't matter what the value of declarationDir.

Should this PR include logic which handles this case?

> That is a JSON feature really. we would be happy to take a PR to the docs to make that clear as well.
> VS Code warns about null: Incorrect Type, expected "string".

`declarationDir` has the type - _optional string_(`declarationDir?: string;`), that's mean that TS compiler throws an error about incompatible type if the value type is not `""` or `undefined`, so the following example will not work in TS. 

```typescript
const parsedOptions2 = ts.parseJsonConfigFileContent(optionsFromTsconfig, host, "./", { declaration: false, declarationDir: null });
```
however, in JS/JSON we can use `null`. In order to use `null` as value _(in TS)_, need to change type, I suppose it should look like so

```typescript
declarationDir?: string | null;
```
and also, change scheme types for VSCode.

Is this issue related only to `declarationDir`? Or there are other properties which cause similar errors?

**Comment by jakebailey** [comment]
Just looking at old issues; this one was fixed in [`869abb1` (#39243)](https://github.com/microsoft/TypeScript/pull/39243/commits/869abb1414a96e9622f5b04166df53b470bc3a88#diff-4dd2736e47d6b51eaf8ff275c70e53dc5f6bca718a54a2af474cd06147413f68R2825).
